front-end/src/Scenes/Registration/Components/RegistrationForm.jsx

```diff
@@ -0,0 +1,425 @@
+      span: 24,
+      offset: 0,
+    },
+    sm: {
+      span: 16,
+      offset: 8,
+    },
+  },
+};
+
+class RegistrationForm extends React.Component {
+  state = {
+    teacher: false,
+    idInstitution: -1,
+    selectedInstitutionTeacher: -1,
+    inputtedJob: '',
+    email: '',
+    firstName: '',
+    secondName: '',
+    faculty: '',
+    primarySkill: '',
+    yearTermination: 0,
+  };
+
+  setTextField = (event) => {
+    const field = event.target.name;
+    const targetValue = event.target.value;
+    this.setState(prevState => ({
+      [field]: prevState[field].replace(
+        prevState[field],
+        targetValue || '',
+      ),
+    }));
+  };
+
+  setYearTermination = (event) => {
+    const { value } = event.target;
+    this.setState(prevState => ({
+      yearTermination: prevState.yearTermination - prevState.yearTermination + value,
+    }));
+  };
+
+  setSelectField = (event, name) => {
+    const field = name;
+    this.setState(prevState => ({
+      [field]: prevState[field] - prevState[field] + event,
+    }));
+  };
+
+  changeSwitch = () => this.setState(prevState => ({ teacher: !prevState.teacher }));
+
+  render() {
+    const {
+      teacher,
+      idInstitution,
+      selectedInstitutionTeacher,
+      inputtedJob,
+      email,
+      firstName,
+      secondName,
+      faculty,
+      primarySkill,
+      yearTermination,
+    } = this.state;
+    const { form, registrationTeacher, registrationStudent } = this.props;
+    const { getFieldDecorator } = form;
+
+
+    const getUser = () => (teacher
+      ? {
+        email,
+        firstName,
+        secondName,
+        institution: selectedInstitutionTeacher !== -1 ? institutions[selectedInstitutionTeacher].name : '',
+        job: inputtedJob,
+        primarySkill,
+      }
+      : {
+        email,
+        firstName,
+        secondName,
+        institution: idInstitution !== -1 ? institutions[idInstitution].name : '',
+        faculty,
+        yearTermination,
+        primarySkill,
+      });
+
+    const switchRegistration = () => (teacher
+      ? registrationTeacher(getUser())
+      : registrationStudent(getUser())
+    );
+
+    const handleSubmit = (e) => {
+      e.preventDefault();
+      form.validateFieldsAndScroll((err) => {
+        if (!err) {
+          switchRegistration();
+        }
+      });
+    };
+
+    const institutionOptions = institutions.map(inst => (
+      <Option value={inst.id} key={inst.id}>
+        {inst.name}
+      </Option>
+    ));
+
+    const primarySkillsOptions = primarySkills.map(skill => (
+      <Option value={skill} key={skill}>
+        {skill}
+      </Option>
+    ));
+
+    const facultiesOptions = idInstitution !== -1 ? faculties[
+      institutions[idInstitution].id].map(elem => (
+        <Option value={elem} key={elem}>
+          {elem}
+        </Option>
+    )) : [];
+
+    const inputForTeachers = (
+      <FormItem>
+        <FormItem {...formItemLayout} label="Место работы">
```

Comment:
i don't know how i can simplify 'inputForTeachers' because of the exsistence of the fields with the different names. ant i can't remove the outer <FormItem> because it provides the validation of my form.